src/assistant/Conversations/Sources/Sources.jsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useRef, useEffect } from 'react'
+import keyBy from 'lodash/keyBy'
+
+import { useQuery, isQueryLoading } from 'cozy-client'
+
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import Chip from 'cozy-ui/transpiled/react/Chips'
+import MultiFilesIcon from 'cozy-ui/transpiled/react/Icons/MultiFiles'
+import RightIcon from 'cozy-ui/transpiled/react/Icons/Right'
+import Box from 'cozy-ui/transpiled/react/Box'
+import Grow from 'cozy-ui/transpiled/react/Grow'
+
+import { buildFilesByIds } from 'assistant/queries'
+import SourcesItem from './SourcesItem'
+
+const Sources = ({ filesWitPath }) => {
+  const [showSources, setShowSources] = useState(false)
+  const { t } = useI18n()
+  const ref = useRef()
+
+  const handleShowSources = () => {
+    setShowSources(v => !v)
+  }
+
+  useEffect(() => {
+    if (showSources) {
+      const sourcesBottom = ref.current.getBoundingClientRect().bottom
+      const innerContainer =
+        document.getElementsByClassName('cozyDialogContent')[0]
+      const innerContainerBottom = innerContainer.getBoundingClientRect().bottom
+      if (sourcesBottom > innerContainerBottom) {
+        ref.current.scrollIntoView(false)
+      }
+    }
+  }, [showSources])
+
+  return (
+    <Box ref={ref} className="u-mt-1-half" pl="44px">
+      <Chip
+        className="u-mb-1"
+        icon={<Icon icon={MultiFilesIcon} className="u-ml-half" />}
+        label={t('assistant.sources', filesWitPath.length)}
+        deleteIcon={
+          <Icon
+            className="u-h-1"
+            icon={RightIcon}
+            rotate={showSources ? 90 : 0}
+          />
+        }
+        clickable
+        onClick={handleShowSources}
+        onDelete={handleShowSources}
+      />
+      <Grow
+        in={showSources}
+        style={{ transformOrigin: '0 0 0' }}
+        mountOnEnter={true}
+        unmountOnExit={true}
+      >
+        <div>
+          {filesWitPath.map((fileWitPath, index) => (
+            <SourcesItem
+              key={`${fileWitPath._id}-${index}`}
+              fileWitPath={fileWitPath}
+            />
+          ))}
+        </div>
+      </Grow>
+    </Box>
+  )
+}
+
+const SourcesWithFolderQuery = ({ files }) => {
+  const folderIds = files.map(file => file.dir_id)
+
+  const folderByIds = buildFilesByIds(folderIds)
+  const { data: folders, ...queryResult } = useQuery(
+    folderByIds.definition,
+    folderByIds.options
+  )
+
+  const isLoading = isQueryLoading(queryResult)
+
+  if (isLoading) return null
+
+  const foldersById = keyBy(folders, '_id')
+
+  const filesWitPath = files.map(file => ({
+    ...file,
+    path: foldersById[file.dir_id].path
+  }))
```

Comment:
I don't get this part, isn't the `path` already present in `io.cozy.files` that have the type `file`?

Reply:
non justement c'est bien le souci, le path n'est que sur les folder

Reply:
On en discute demain mais je t'ai induit en erreur @JF-Cozy

Reply:
un `.getByIds(ids)` ne retourne pas le `path`, là où un `.where({ _id: { $in: ids } })` retourne bien le `path`. J'ai donc changé la requête en conséquence et simplifié le code.